Lib/test/libregrtest/worker.py

```diff
@@ -14,6 +14,9 @@
 
 
 USE_PROCESS_GROUP = (hasattr(os, "setsid") and hasattr(os, "killpg"))
+NEED_TTY = set('''
+    test_ioctl
```

Comment:
There's a couple of other tests that skip some of the test methods/classes when stdout is not a tty: test_builtin, test_curses, test_os, test_shutil at least, from a quick grep. I'm not sure it's going to be maintainable to keep a list like this up to date...

Reply:
One option would be to move "tty" tests in separated test files and disable setsid() if the test name contains `_tty`. For example, we already do that with GUI tests (ex: test_ttk and test_ttk_textonly).